### PR TITLE
[ADD] website_sale_category_alphabetic: Automatic alphabetic website categories

### PR DESCRIPTION
**Pull request by obulkin**
* Overload `create` in `product.template` to automatically add alphabetic `product.public.category` records to new products. New categories are created at this time if needed
* Add post-install hook that similarly categorizes all pre-existing `product.template` records
* Add data, demo data, and tests as needed

### CODE REVIEW

File: website_sale_category_alphabetic/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from openerp import api, models
```
Comment (lasley) [inline website_sale_category_alphabetic/models/product_template.py]: `from odoo`

File: website_sale_category_alphabetic/tests/test_hooks.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import mock
+from openerp.tests.common import TransactionCase
```
Comment (lasley) [inline website_sale_category_alphabetic/tests/test_hooks.py]: odoo

**Review by lasley — APPROVED** [review]
Few comments, otherwise LGTM

**Comment by obulkin** [comment]
I've pushed up a new commit that replaces every reference to `openerp` with `odoo`.

**Review by lasley — APPROVED** [review]
Thanks @obulkin

**Comment by lasley** [comment]
FYI the CodeCov error looks to be false positive - this PR has 100% test coverage.

File: website_sale_category_alphabetic/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+{
+    'name': 'Website Sale - Alphabetic Categories',
+    'summary': 'Automatically adds the correct alphabetic website category to'
+               ' all new products',
+    'version': '10.0.1.0.0',
+    'category': 'Website',
+    'website': 'https://laslabs.com/',
+    'author': 'LasLabs',
```
Comment (woodbrettm) [inline website_sale_category_alphabetic/__manifest__.py]: Should this include OCA as well?

**Review by woodbrettm — APPROVED** [review]
LGTM

**Comment by obulkin** [comment]
Good catch, @bmw95. I've added OCA as an author.

**Review by woodbrettm — APPROVED** [review]
Woot! Okie looks good to me :)

**Pull request merged**